grm-db-experiments/debian/manpage.1
.\"                                      Hey, EMACS: -*- nroff -*-
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH GOURMET SECTION "August 21, 2004"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
gourmet \- a recipe manager

This manual page was written for the Debian distribution
because the original program does not have a manual page.

This manual page was written by Thomas Mills Hinkle <Thomas_Hinkle@alumni.brown.edu>,
for the Debian project (but may be used by others).


CHANGES
==== Version 0.13.0 changes since 0.12.2 ====
* Addition of hash fields to help with identification of duplicate
  recipes.
* Addition of last_modified field to recipe database.
* Addition of link field to database - now recipes imported from the
  web will automatically have a link pointing back to the original page.

==== Version 0.12.2 changes since 0.12.1 ====
* Fix Mealmaster importing bug
* Fix bug (?) which could cause ingredient search to fail in strange
  cases (where a DB had ingredients with key=None)

==== Version 0.12.1 changes since 0.12.0-pre1 ====

* Fix buggy trash behavior.
* Store version in database for ease of upgrades.
* Implement fix for broken databases from pre-0.11.4 installs
* Fix-up paragraphing in printing.

==== Version 0.12.0-pre1 changes since 0.11.5 ====
* Fix sort order
* Don't drop default entries for Category/Cuisine once user has entered one.
* Change version number to reflect intend to create a stable version
  once we're sure this version doesn't have any show-stoppers.

==== Version 0.11.5 changes since 0.11.4 ====
* Add more PDF Export modes for printing index cards, multiple
  columns, etc. Also add nice print page-layout widget.
* Tweak spacing in recipe card to avoid ugly disappearance of "Edit" buttons
* Recognize units in all kinds of capitalized or uncapitalized forms.
* Fix up bug in time-scanning code that led to making a link like
  <a>3 min</a>utes instead of <a>3 minutes</a>
* Add i18n for Swedish.
* Apply patch to fix text-flow in recipe card (Shane Hathaway)
* Apply localization patch to simplify translations. (Jens Persson)

==== Version 0.11.4 changes since 0.11.3 ====
* Make deleting an ingredient ungrey "Save" button
* Fix various bits of badness with Undo
* Fix gravity of cursor for text markup in instructions and notes.
* Fix allrecipes.com import
* Allow recognition of times written as 01:30 for 1 1/2 hours
* Improve testing suites (and therefore improve thoroughness of
  bug-checking before release)
* Fixed generic web import.
* PDF Import
* Printing support for windows
* Fix ugly window placement issues on windows and some other places.
* Improve recipezaar import.

==== Version 0.11.3 changes since 0.11.2 ====
* i18n fixes from Simon Darlington
* Fix bug that reversed ingredient order when adding ingredients to a group.
* Allow fraction-use to be turned off, and make it default to off for
  German (other locales can default it to off via defaults_foo.py)
* Fix bug that caused unicode error on upgrade from metakit.
* Fix some other bugginess in handling old upgrades.
* Rearrange main index window to work better on small screens.
* Update to DTD thanks to Gianpaolo Racca.

==== Version 0.11.2 changes since 0.11.1 ====

* Add new icons from Yula Zubritsky
* Add nifty timer.
* Parse times written out as words.

==== Version 0.11.1 changes since 0.11.0 ====

* Fix bug that broke changing the shopping category of an ingredient
  from within the ingredient editing view.
* Fix traceback when filling ingredient-editor from a unitless
  ingredient.
* Fix problem with recipe-links not working when they contained
  unicode (such as the unicode for 1/2, 1/4, etc.)
* Make page-chooser in edit-view highlight the first page when we open
  a new recipe card. Otherwise, the function of the widget is unclear.

==== Version 0.11.0 changes since 0.10.1 ====

* Fix ingredient editing so that changes are no longer automatically
  committed to the DB. Now the ingredient tab of the recipe edit window
  works like everything else, with a standard "save" model.
* While we're at it, we prevent crashes from happening.
* Make Undo work more consistently throughout recipe card.
* Make database upgrade happen at runtime rather than install-time --
  this allows us to play nicely with RPM and DEB packages.

==== Version 0.10.2 changes since 0.10.1 ====

* Fix bug causing export from recipe card not to multiply ingredients
  properly.
* Fix various problems with ingredient editing (display of updates,
  undo/redo)
* Fix dialog responses so hitting return and escape works more consistently.
* Fix bug in addition of ingredients in shopping list.

==== Version 0.10.1 changes since 0.10.0 ====
* Fix bug causing recipe cards to be destroyed when closed.
* Fix bug causing "None" to show up in shopping list when there is no
  unit.
* Fix popup menus throughout.

==== Version 0.10.0 changes cince 0.9.0.3 ====
* Switch database backend to SQLite. The backend is still abstract, so
  if in the future someone wants to switch back to metakit or to
  something else, it should be simple.
* Add new automated backup script into setup.py. This is less than
  ideal, but there's no real way to import smoothly from the old system
  to the new.
* Make editing of recipe card a separate dialog from viewing recipe
  card. This allows the recipe card itself to lose the toolbar and look
  much cleaner.
* Make text in recipe card reflow properly when resizing.
* Improve import speed drastically by not inserting auto-generated
  keys into key table.
* Only add keys to ingkey table (used for automatic key-guessing) when
  the user has modified something. This means we actually learn from the
  user rather than generating our own random noise.
* Add support for typing "or" in a search instead of using the or bar (|).
* Import sorting of USDA DB searches (in the nutritional info druid)
  to bring you better results faster.
* Many, many more bugfixes along the way.

 
==== Version 0.9.0.3 changes since 0.9.0.1 ====
* Fix email export
* Improve recipe card interface.

==== Version 0.9.0.2 changes since 0.9.0.1 ====
* Support python2.3

==== Version 0.9.0.1 changes since 0.9.0 ====
* Fixed shopping list handling of ranges.
* Fixed up MANIFEST to include all necessary files in our release.

==== Version 0.9.0 Changes since 0.8.x ====
* Implemented nutritional information database.
* Fixed a number of bugs causing crashing with the pageable-list view / metakit

grm-db-experiments/debian/manpage.sgml.ex
<!doctype refentry PUBLIC "-//OASIS//DTD DocBook V4.1//EN" [

<!-- Process this file with docbook-to-man to generate an nroff manual
     page: `docbook-to-man manpage.sgml > manpage.1'.  You may view
     the manual page with: `docbook-to-man manpage.sgml | nroff -man |
     less'.  A typical entry in a Makefile or Makefile.am is:

manpage.1: manpage.sgml
	docbook-to-man $< > $@

    
	The docbook-to-man binary is found in the docbook-to-man package.
	Please remember that if you create the nroff version in one of the
	debian/rules file targets (such as build), you will need to include
	docbook-to-man in your Build-Depends control field.

  -->

  <!-- Fill in your name for FIRSTNAME and SURNAME. -->
  <!ENTITY dhfirstname "<firstname>FIRSTNAME</firstname>">
  <!ENTITY dhsurname   "<surname>SURNAME</surname>">
  <!-- Please adjust the date whenever revising the manpage. -->
  <!ENTITY dhdate      "<date>August 21, 2004</date>">
  <!-- SECTION should be 1-8, maybe w/ subsection other parameters are
       allowed: see man(7), man(1). -->
  <!ENTITY dhsection   "<manvolnum>SECTION</manvolnum>">
  <!ENTITY dhemail     "<email>Thomas_Hinkle@alumni.brown.edu</email>">
  <!ENTITY dhusername  "Thomas Mills Hinkle">
  <!ENTITY dhucpackage "<refentrytitle>GOURMET</refentrytitle>">
  <!ENTITY dhpackage   "gourmet">

  <!ENTITY debian      "<productname>Debian</productname>">
  <!ENTITY gnu         "<acronym>GNU</acronym>">
  <!ENTITY gpl         "&gnu; <acronym>GPL</acronym>">
]>

<refentry>
  <refentryinfo>
    <address>
      &dhemail;
    </address>
    <author>
      &dhfirstname;
      &dhsurname;
    </author>
    <copyright>
      <year>2003</year>
      <holder>&dhusername;</holder>
    </copyright>
    &dhdate;
  </refentryinfo>
  <refmeta>
    &dhucpackage;

    &dhsection;
  </refmeta>
  <refnamediv>
    <refname>&dhpackage;</refname>

    <refpurpose>program to do something</refpurpose>
  </refnamediv>
  <refsynopsisdiv>
    <cmdsynopsis>
      <command>&dhpackage;</command>

      <arg><option>-e <replaceable>this</replaceable></option></arg>

      <arg><option>--example <replaceable>that</replaceable></option></arg>
    </cmdsynopsis>
  </refsynopsisdiv>
  <refsect1>
    <title>DESCRIPTION</title>

    <para>This manual page documents briefly the
      <command>&dhpackage;</command> and <command>bar</command>
      commands.</para>

    <para>This manual page was written for the &debian; distribution
      because the original program does not have a manual page.
      Instead, it has documentation in the &gnu;
      <application>Info</application> format; see below.</para>

    <para><command>&dhpackage;</command> is a program that...</para>

  </refsect1>
  <refsect1>
    <title>OPTIONS</title>

    <para>These programs follow the usual &gnu; command line syntax,
      with long options starting with two dashes (`-').  A summary of
      options is included below.  For a complete description, see the
      <application>Info</application> files.</para>

    <variablelist>
      <varlistentry>
        <term><option>-h</option>
          <option>--help</option>
        </term>
        <listitem>
          <para>Show summary of options.</para>
        </listitem>
      </varlistentry>
      <varlistentry>
        <term><option>-v</option>
          <option>--version</option>
        </term>
        <listitem>
          <para>Show version of program.</para>
        </listitem>
      </varlistentry>
    </variablelist>
  </refsect1>
  <refsect1>
    <title>SEE ALSO</title>

    <para>bar (1), baz (1).</para>

    <para>The programs are documented fully by <citetitle>The Rise and
      Fall of a Fooish Bar</citetitle> available via the
      <application>Info</application> system.</para>
  </refsect1>
  <refsect1>
    <title>AUTHOR</title>

    <para>This manual page was written by &dhusername; &dhemail; for
      the &debian; system (but may be used by others).  Permission is
      granted to copy, distribute and/or modify this document under
      the terms of the &gnu; General Public License, Version 2 any 
	  later version published by the Free Software Foundation.
    </para>
	<para>
	  On Debian systems, the complete text of the GNU General Public
	  License can be found in /usr/share/common-licenses/GPL.
	</para>

  </refsect1>
</refentry>

<!-- Keep this comment at the end of the file
Local variables:
mode: sgml
sgml-omittag:t
sgml-shorttag:t
sgml-minimize-attributes:nil
sgml-always-quote-attributes:t
sgml-indent-step:2
sgml-indent-data:t
sgml-parent-document:nil
sgml-default-dtd-file:nil
sgml-exposed-tags:nil
sgml-local-catalogs:nil
sgml-local-ecat-files:nil
End:
-->


